### PR TITLE
Allow NULL values for CSV exports

### PR DESCRIPTION
**Pull request by alex-mo**
See also https://www.drupal.org/node/2840930

### CODE REVIEW

File: webform_currency.component.inc [hunk 1]
```diff
@@ -307,17 +307,20 @@ function _webform_csv_headers_currency(array $component, array $export_options)
...
  *
  * @return string
  *   An array of items to be added to the CSV file. Each value within the array
  *   will be another column within the file. This function is called once for
  *   every row of data.
  */
-function _webform_csv_data_currency(array $component, array $export_options, array $value) {
-  return isset($value[0]) ? $value[0] : '';
+function _webform_csv_data_currency(array $component, array $export_options, $value) {
+  return $value && isset($value[0]) ? $value[0] : '';
```
Comment (katzmo) [inline webform_currency.component.inc]: I think `return $value[0] ?? ''` would do the same and be a lot easier to read? :wink:

Comment (torotil) [reply webform_currency.component.inc]: @katzmo We’d need to bump the PHP requirement then though. Usually this means tagging a major release.

Comment (katzmo) [reply webform_currency.component.inc]: I see. Then `return !empty($value[0]) ?  $value[0] : '';` would work?

Comment (torotil) [reply webform_currency.component.inc]: How is this different in terms of warnings/notices? Locally for me even the orginal version with `isset()` seems to work so I can’t test this.

Comment (torotil) [reply webform_currency.component.inc]: Ah I see. The essential part is the type-hint in the function declaration. The change in the function body is actually not needed.

```php
$value = NULL;
isset($value[0]);
```
works as it is. (PHP 7.3)

Comment (alex-mo) [reply webform_currency.component.inc]: changed

**Review by torotil — APPROVED** [review]

**Pull request merged**